src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -55,7 +64,11 @@ public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyDeliveryJobSystem d
         this.filteredPersons = new FilteredList<>(this.addressBook.getPersonList());
         this.filteredDeliveryJobs = new FilteredList<>(this.deliveryJobSystem.getDeliveryJobList());
         this.sortedDeliveryJobs = new ArrayList<DeliveryJob>(this.deliveryJobSystem.getDeliveryJobList());
+        //updateSortedDeliveryJobListByDate();
+        this.jobListGroupedByDate = getSortedDeliveryJobListByDate();
+        this.weekJobListGroupedByDate = new HashMap<LocalDate, ArrayList<ArrayList<DeliveryJob>>>();
         this.reminderList = this.addressBook.getReminderList();
+        this.focusDate = LocalDate.now();
```

Comment:
focusDate seems to be a front-end variable, i'm not sure if its suitable to place it in data layer.

Reply:
i left it there cuz i think i may need it later, cuz i wanted to do like, when users key in a date then the timetable shows the week containing that date and the tasks. So focusDate is basically used to change the timetable to the corresponding week